server/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java

```diff
@@ -360,9 +359,7 @@ protected void onNodeResponse(DiscoveryNode node, int nodeIndex, NodeResponse re
 
         protected void onNodeFailure(DiscoveryNode node, int nodeIndex, Throwable t) {
             String nodeId = node.getId();
-            if (logger.isDebugEnabled() && (t instanceof NodeShouldNotConnectException) == false) {
-                logger.debug(new ParameterizedMessage("failed to execute [{}] on node [{}]", actionName, nodeId), t);
-            }
+            logger.debug(new ParameterizedMessage("failed to execute [{}] on node [{}]", actionName, nodeId), t);
```

Comment:
NIT: maybe keep the `logger.isDebugEnabled()` here and in the other spot so that we don't have to create the message object needlessly?